src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -133,26 +150,28 @@ public Command intake() {
 			elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).handleInterrupt(() -> enableChangeStateAutomatically(true).schedule());
 	}
 
 	public Command preSpeaker() {
 		return new ParallelCommandGroup(
 			setCurrentStateName(RobotState.PRE_SPEAKER),
+            enableChangeStateAutomatically(true),
 //			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER)),
 			elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
 			flywheelStateHandler.setState(FlywheelState.SHOOTING),
 			funnelStateHandler.setState(FunnelState.STOP),
 			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).handleInterrupt(() -> enableChangeStateAutomatically(true).schedule());
 	}
 
 	public Command speaker() {
 		return new ParallelCommandGroup(
 			setCurrentStateName(RobotState.SPEAKER),
-			new SequentialCommandGroup(
+            enableChangeStateAutomatically(false),
```

Comment:
true after note is out?

Reply:
Where would you put that? I don't think it's needex

Reply:
You want to return to idle after you shoot
you can put it after funnel stops the second time

Reply:
It's similar to what's happening in intake